### PR TITLE
Code cleanliness improvements

### PR DESCRIPTION
**Pull request by Calvinp**
- Renames most one and two letter variable names to things that are more descriptive.
- Adds an eslint rule to disallow most one and two letter variable names. `id`, `ui`, and `s3` are ok because they are useful and informative when they are used, unlike `hc` (for `healthCheck`)
- Adds a block exception to this rule for the moment locale rules.
- Fixes a spot where props were mutated.
- Uses `moment.updateLocale()` instead of `moment.locale()` to get rid of the deprecated warning.
- Makes several components functional.

### CODE REVIEW

**Comment by Calvinp** [comment]
I always forget to do this... cc @tpetr @kwm4385 @wolfd

File: SingularityUI/app/components/common/formItems/Duration.jsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default class Duration extends React.Component {
             min={0}
             step={1}
             value={this.props.value ? duration.hours() : ''}
-            onChange={(e) => this.handleChange(e)}
+            onChange={(event) => this.handleChange(event)}
```
Comment (kwm4385) [inline SingularityUI/app/components/common/formItems/Duration.jsx]: These could probably be shortened just to `onChange={this.handleChange}`

File: SingularityUI/app/components/machines/Racks.jsx [hunk 2]
```diff
@@ -17,49 +17,29 @@ const typeName = {
...
-        <p>Reactivating a rack will cancel the decommission without erasing the rack's history and move it back to the active state.</p>
-      </ModalButton>
-    );
-  }
-
-  getDecommissionOrRemoveButton(rack) {
+const Racks = (props) => {
+  const showUser = (rack) => Utils.isIn(rack.currentState.state, ['ACTIVE', 'DECOMMISSIONING', 'DECOMMISSIONED', 'STARTING_DECOMMISSION']);
+
+  const getMaybeReactivateButton = (rack) => (Utils.isIn(rack.currentState.state, ['DECOMMISSIONING', 'DECOMMISSIONED', 'STARTING_DECOMMISSION']) &&
```
Comment (kwm4385) [inline SingularityUI/app/components/machines/Racks.jsx]: Might want to split these into their own functions, its difficult to follow what's going on here.

File: SingularityUI/app/components/common/formItems/Duration.jsx (hunk 1 above)
Comment (Calvinp) [inline SingularityUI/app/components/common/formItems/Duration.jsx]: When making this change, I found a bug (that is present in `decaf` too). I'm working on fixing that.

File: SingularityUI/app/components/dashboard/MyPausedRequests.jsx [hunk 3]
```diff
@@ -13,13 +13,13 @@ const MyPausedRequests = ({userRequests}) => {
...
 
-  const pausedRequests = userRequests.filter((r) => r.state === 'PAUSED');
+  const pausedRequests = userRequests.filter((request) => request.state === 'PAUSED');
 
   if (pausedRequests.length > 0) {
     pausedRequestsSection = (
       <UITable
         data={pausedRequests}
-        keyGetter={(r) => r.request.id}
+        keyGetter={(request) => request.request.id}
```
Comment (wolfd) [inline SingularityUI/app/components/dashboard/MyPausedRequests.jsx]: these are actually requestParents, but we don't do a good job of labeling them...

Comment (Calvinp) [inline SingularityUI/app/components/dashboard/MyPausedRequests.jsx]: Yeah, I'm realizing that. I'm willing to go through and change them all - it's probably worth doing.

File: SingularityUI/app/selectors/requests.es6 [hunk 4]
```diff
@@ -1,7 +1,6 @@
 import { createSelector } from 'reselect';
 import micromatch from 'micromatch';
 import fuzzy from 'fuzzy';
-import _ from 'underscore';
```
Comment (tpetr) [inline SingularityUI/app/selectors/requests.es6]: i'd like to move away from global imports, not towards -- can we add this back?

File: SingularityUI/app/utils.es6 [hunk 5]
```diff
@@ -132,16 +132,16 @@ const Utils = {
     }
   },
 
-  joinPath(a, b) {
-    if (!a.endsWith('/')) a += '/';
-    if (b.startsWith('/')) b = b.substring(1, b.length);
-    return a + b;
+  joinPath(firstPart, secondPart) {
+    if (!firstPart.endsWith('/')) firstPart += '/';
+    if (secondPart.startsWith('/')) secondPart = secondPart.substring(1, secondPart.length);
```
Comment (wolfd) [inline SingularityUI/app/utils.es6]: substring has the second arg optional if you want it to go to the end.

**Pull request merged**